src/gsi/hybrid_ensemble_isotropic.F90

```diff
@@ -3711,60 +3723,133 @@ subroutine bkgcov_a_en_new_factorization(ig,a_en)
   type(gsi_bundle),intent(inout) :: a_en(n_ens)
 
 ! Local Variables
-  integer(i_kind) ii,k,iadvance,iback,is,ie,ipnt,istatus
+  integer(i_kind) ii,i,j,k,iadvance,iback,is,ie,ipnt,istatus,kk
   real(r_kind) hwork(grd_loc%inner_vars,grd_loc%nlat,grd_loc%nlon,grd_loc%kbegin_loc:grd_loc%kend_alloc)
+  real(r_kind) work_tmp(grd_loc%lon2,grd_loc%lat2)
   real(r_kind),allocatable,dimension(:):: a_en_work
 
   ipnt=1
 
 ! Apply vertical smoother on each ensemble member
+  if(regional.and.l_mgbf_loc.and.ig==1) then
```

Comment:
@shoyokota why is ig==1 required ?  The mgbf localization is applicable to multiscale runs? 
I understand l_mbgf is only tested for regional application now and hence, "regional" is one of the conditions. 
But I would suggestion , at an earlier place like in gsimod.f90, to make sure only regional=.true., l_mbgf is slected.
Here, "regional" could be removed from the conditions to make the codes easier to understand.

Reply:
@TingLei-NOAA To emphasize that MGBF is not applicable for global application, I didn't remove "regional" here, but I added the process to force l_mgbf_loc=false for global application.

MGBF-localization is applicable also with SDL/VDL. In SDL/VDL, this subroutine (bkgcov_a_en_new_factorization) is called only for scale-separation (ig>naensgrp), and localization (ig<=naensgrp) is computed in other subroutines (ckgcov_a_en_new_factorization and ckgcov_a_en_new_factorization_ad). Not to use MGBF for scale-separation, I added ig==1 here in bkgcov_a_en_new_factorization. (Investigation about applying MGBF for scale-separation is one of the future works.)

Reply:
@shoyokota  I would have some further discussion on the use of l_mgbf_loc together with other options.  The use of "if(regional.and.l_mgbf_loc.and.ig==1) then " doesn't help to emphasize l_mgbf_loc only work for regional , partially because ig==1 also exist.  Another example, like Line 4513 " if( .not. l_mgbf_loc .or. nsclgrp > 1 ) then"   is also confusing compared with previous version.

Reply:
Also, I am still wondering why ig>1 should be taken care in other places.  Why could they not be done through this block ?  The logics with the sdl/vdl had already brought confusion to original codes.  Adding of l_mgbf_loc caused more questions to follow the logics. Hope some simple modifications could work.  Since this PR is mainly for 3D RTMA , for being now.  The sdl/vdl is not the necessary option for it and also, I think those functions hasn't been tested for sdl/vdl, I would recommend assuming l_mbgf_loc only work for non-sdl/vdl cases and, simplify those logics.

Reply:
The functions for MGBF with SDL/VDL already have been tested and presented in [AMS2024](https://github.com/NOAA-EMC/GSI/files/14499956/20240129_Yokota_MGBF-loc.pdf), and in my understanding, it hasn't been determined if SDL/VDL is used for 3DRTMA yet. @ManuelPondeca-NOAA, could I ask your opinion if I should remove the functions for MGBF-based SDL/VDL or not?

In this MGBF-based SDL/VDL, Recursive Filter is still used only in the scale separation. This implementation is based on the idea that the scale separation should be done by (quasi-)Gaussian Filter. That's why ig>1 should be taken care in the parts related to Recursive Filter (including line 4513). If necessary, I will add some comments in such parts.

Reply:
@shoyokota I am still reading mgbf related changes. but now,  I have a recommendation.  In the current  codes. 
It is organized as "  { do vertical filter { if mgbf , do mgbf vertical filer ; else , original recursive filter};  do horizontal smoother { if mgbf, do mgbf smoother, else, original recursive filter} ..... }" , I would suggest to put mgbf vs rf in the most outer. namely "{if mgbf , do mgbf {vertical filter, horizontal smoother}, else do original recursive filter{do vertical filter, do horizontal smoother}.}.  In my opinion, it is a better organized structure with better  readability. Also , It will not prevent using vertical RF in the mgbf branch.

Reply:
A clarification, such re-organization of the branching of   mgbf vs RF  is limited to each related subs

Reply:
I re-organized the code as you suggested, and added some comments to make it easier to understand. Is the updated code more readable?